ui/apps/platform/src/Containers/ComplianceEnhanced/ScanConfigs/Table/ScanConfigsTablePage.tsx

```diff
@@ -1,98 +1,248 @@
+import EmptyStateTemplate from 'Components/PatternFly/EmptyStateTemplate';
+import useRestQuery from 'hooks/useRestQuery';
+import useURLPagination from 'hooks/useURLPagination';
+import useURLSort from 'hooks/useURLSort';
+import { getScanConfigs, Schedule } from 'services/ComplianceEnhancedService';
+import { SortOption } from 'types/table';
+import { displayOnlyItemOrItemCount } from 'utils/textUtils';
+import { getDayOfMonthWithOrdinal, getTimeHoursMinutes } from 'utils/dateUtils';
+
 import ScanConfigsHeader from '../ScanConfigsHeader';
 
 type ScanConfigsTablePageProps = {
     hasWriteAccessForCompliance: boolean;
-    handleChangeSearchFilter: (searchFilter: SearchFilter) => void;
-    searchFilter?: SearchFilter;
 };
 
+const CreateReportsButton = () => {
+    return (
+        <Link to={`${complianceEnhancedScanConfigsBasePath}/?action=create`}>
+            <Button variant="primary">Create scan schedule</Button>
+        </Link>
+    );
+};
+
+const sortFields = ['Name', 'Last Run'];
+const defaultSortOption = { field: 'Name', direction: 'asc' } as SortOption;
+
 function ScanConfigsTablePage({
     hasWriteAccessForCompliance,
-    handleChangeSearchFilter,
-    searchFilter,
 }: ScanConfigsTablePageProps): React.ReactElement {
-    const history = useHistory();
-
-    const [isLoading, setIsLoading] = useState(false);
-    const [scanSchedules, setScanSchedules] = useState<ScanConfig[]>([]);
-    const [errorMessage, setErrorMessage] = useState('');
-
-    const [searchOptions, setSearchOptions] = useState<string[]>([]);
-
-    function onClickCreate() {
-        history.push(`${complianceEnhancedScanConfigsBasePath}/?action=create`);
-    }
-
-    function fetchScanSchedules(query: string) {
-        setIsLoading(true);
-        getScanConfigs(query)
-            .then((data) => {
-                setScanSchedules(data);
-                setErrorMessage('');
-            })
-            .catch((error) => {
-                setScanSchedules([]);
-                setErrorMessage(getAxiosErrorMessage(error));
-            })
-            .finally(() => setIsLoading(false));
-    }
-
-    const query = searchFilter ? getRequestQueryStringForSearchFilter(searchFilter) : '';
-
-    useEffect(() => {
-        fetchScanSchedules(query);
-    }, [query]);
-
-    let pageContent = (
-        <PageSection variant="light" isFilled id="policies-table-loading">
-            <Bullseye>
-                <Spinner isSVG />
-            </Bullseye>
-        </PageSection>
+    const { page, perPage, setPage, setPerPage } = useURLPagination(10);
+    const { sortOption, getSortParams } = useURLSort({
+        sortFields,
+        defaultSortOption,
+    });
+
+    const listQuery = useCallback(
+        () => getScanConfigs(sortOption, page - 1, perPage),
+        [sortOption, page, perPage]
     );
+    const { data: scanSchedules, loading: isLoading, error } = useRestQuery(listQuery);
+
+    const formatScanSchedule = (schedule: Schedule) => {
```

Comment:
Is this worth moving to `dateUtils.ts`?

Reply:
Looks like we can can combine this logic with whats in `ScheduleDetails.tsx` for Vulnerability Reporting. I'll sync with @sachaudh on that.

Reply:
I'm going to do that in a follow up PR. It won't be as easy as pulling it out of here and importing it in both places given the strings involved.